extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java

```diff
@@ -32,6 +33,9 @@
 @Recorder
 public class HibernateValidatorRecorder {
 
+    private static final TypeLiteral<ValueExtractor<?>> TYPE_LITERAL_VALUE_EXTRACTOR_WITH_WILDCARD = new TypeLiteral<ValueExtractor<?>>() {
+    };
```

Comment:
This is the only way I found to select all beans implementing `ValueExtractor<Something>`.

Strangely, `.select(ValueExtractor.class)` does not match a bean implementing `ValueExtractor<List<?>>`, for example.

I say strangely, because you can definitely assign `ValueExtractor<List<?>>` to a variable of type `ValueExtractor` (the raw type). So if you consider that `.select(ValueExtractor.class)` should select all beans that can be assigned to `ValueExtractor` (the raw type), then beans implementing `ValueExtractor<List<?>>` should definitely be selected.

But no, that's not how it works, apparently.

According to the javadoc on `io.quarkus.arc.impl.BeanTypeAssignabilityRules#matches(java.lang.Class<?>, java.lang.reflect.ParameterizedType)`:

```
     * A parameterized bean type is considered assignable to a raw required type if the raw types are identical and all type
     * parameters of the bean type are
     * either unbounded type variables or java.lang.Object.
```

So according to this definition, `ValueExtractor<T>` is assignable to `ValueExtractor`, but `ValueExtractor<List<?>>` is not.

@stuartwdouglas is this on purpose? Or should we adjust the implementation of `io.quarkus.arc.impl.BeanTypeAssignabilityRules#matches(java.lang.Class<?>, java.lang.reflect.ParameterizedType)`?

Reply:
This is a question for @mkouba, but this sounds like a bug.

Reply:
Ok, sorry for the noise. Created https://github.com/quarkusio/quarkus/issues/20410 for @mkouba  :)

Reply:
This works as required by the spec: _"A parameterized bean type is considered assignable to a raw required type if the raw types are identical and all type parameters of the bean type are either unbounded type variables or java.lang.Object"_; see also https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#assignable_parameters